openstates/in/legislators.py

```diff
@@ -1,111 +1,47 @@
-            return
-        leg = Legislator(term, chamber, district, name, party=party)
-
-        # Scrape leg page.
-        try:
-            html = self.urlopen(url)
-        except scrapelib.HTTPError as exc:
-            # As of July 2014, this only happens when a page has
-            # gone missing from their varnish server.
-            # if exc.response.status_code is 503:
-            self.logger.exception(exc)
-            self.logger.warning('Skipping legislator at url: %s' % url)
-            skipped = True
-            return
-
-        doc = lxml.html.fromstring(html)
-        doc.make_links_absolute(self.url)
-        leg.add_source(url)
-
-        # Scrape committees.
-        for tr in doc.xpath('//div[@class="legislator-committees-container"]//table//tr'):
-            committee, committee_type, role = tr
-            committee = committee.text_content().strip()
-            role = role.text_content().strip()
-            if 'member' in role.lower():
-                role = 'committee member'
-            elif 'chair' in role.lower():
-                role = 'chair'
-            if committee != "Committee Name":
-                leg.add_role(role, term, chamber=chamber, committee=committee)
-
-        # Scrape offices.
-        dist_office, phone = doc.xpath('//address')
-        dist_office = dist_office.text_content().strip()
-        dist_office = re.sub(r' {2,}', '', dist_office)
-
-        phone = phone.text_content().strip()
-        email = doc.xpath('string(//a[starts-with(@href, "mailto:")]/@href)')
-        photo_url = doc.xpath('string(//img[contains(@class, "member")]/@src)')
-
-        leg.update(email=email, photo_url=photo_url)
-        leg.add_office(
-            address=dist_office, name='Capitol Office',
-            type='capitol', phone=phone)
+        client = ApiClient(self)
+        t = next((item for item in self.metadata["terms"] if item["name"] == term),None)
+        session = max(t["sessions"])
+        base_url = "https://iga.in.gov/legislative"
+        api_base_url = "https://api.iga.in.gov/{}".format(session)
+        chamber_name = "Senate" if chamber == "upper" else "House"
+        r = client.get("chamber_legislators",session=session,chamber=chamber_name)
+        all_pages = client.unpaginate(r)
+        for leg in all_pages:
+            print leg
+            firstname = leg["firstName"]
+            lastname = leg["lastName"]
+            party = leg["party"]
+            link = leg["link"]
+            api_link = api_base_url+link
+            html_link = base_url+link.replace("legislators/","legislators/legislator_")
+            html = get_with_increasing_timeout(self,html_link,fail=True,kwargs={"verify":False})
+            doc = lxml.html.fromstring(html.text)
+            doc.make_links_absolute(html_link)
+            address, phone = doc.xpath("//address")
+            address = address.text_content().strip()
+            phone = phone.text_content().strip()
```

Comment:
This `phone` data isn't added to the legislator. I don't think it'd be inappropriate to add an otherwise empty office (no address, no email, no fax, just phone), right?